api/src/main/java/com/lifegraph/team20/controllers/SearchController.java

```diff
@@ -0,0 +1,73 @@
+package com.lifegraph.team20.controllers;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.lifegraph.team20.models.SearchGraphInfo;
+
+@RestController
+public class SearchController {
+/**
+ *  LikeNameであいまい検索ができるよう設定
+ * 現在は、仮設定中
+ * static final String で中の文字が含まれるデータが全て引っ張ってこられる。
+ *　スラッシュでコメントアウトしているコードはバックで特定の引数をSQLから引っ張るようのAPIテスト用のコードなので無視
+ */
+//	private static final String LikeName = "私";
+//	private static final String StartDate = "";
+//	private static final String UpdateDate = "";
+
+	@RequestMapping(value = "/auth/search", method = RequestMethod.GET)
+// SearchGraphInfoから検索ページに必要な要素を引っ張ってきて、反映。 l.33はl.50に影響している。　　　like_nameを引数にしてあいまい検索を行えるようにしている。　バックだけでは引数"私"のようなものがない為、フロントと繋げる必要あり。
+	public List<SearchGraphInfo>SearchGraphInfo(@RequestParam("LikeName") Optional<String>like_name,
+												@RequestParam("StartDate") Optional<Timestamp>start_date,
+												@RequestParam("UpdateDate") Optional<Timestamp>update_date){
+		List<SearchGraphInfo> SearchGraphInfos = SelectSearchGraphInfo(like_name,start_date,update_date);
+		return SearchGraphInfos;
+	}
+//	public List<SearchGraphInfo>SearchGraphInfo(){
+//			List<SearchGraphInfo> SearchGraphInfos = SelectSearchGraphInfo();
+//			return SearchGraphInfos;
+//	}
+// 日時検索のためには、変更、編集が必要かも？
+	@Autowired
+	private JdbcTemplate jdbcTemplate;
+	private List<SearchGraphInfo> SelectSearchGraphInfo(Optional<String> like_name, Optional<Timestamp> start_date, Optional<Timestamp> update_date){
+		String sql="";
+		if(like_name != null) {
```

Comment:
Optionalの比較でnull比較を行なっても、条件分岐が正しくないです。
```suggestion
		if(like_name.isPresent()) {
```